ShovIP/app/src/main/java/com/example/shovip/DialPadFragment.kt

```diff
@@ -0,0 +1,76 @@
+package com.example.shovip
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.navigation.fragment.findNavController
+import com.example.shovip.databinding.FragmentDialPadBinding
+
+/**
+ * A simple [Fragment] subclass as the default destination in the navigation.
+ */
+class DialPadFragment : Fragment() {
+    private var binding by autoCleared<FragmentDialPadBinding>()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentDialPadBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    @SuppressLint("SetTextI18n")
+    private fun loadData(){
+        val act = this.activity
+        val sharedPreferences = act!!.getSharedPreferences("sharedPrefs", Context.MODE_PRIVATE)
```

Comment:
@shovian In general, using the `!!` operator is never a good idea unless you absolutely have to. Please check if this can be avoided.

Reply:
Here's a suggestion using `?.let` syntax:

```
    private fun loadData(){
        activity?.let {
            val sharedPreferences = it.getSharedPreferences("sharedPrefs", Context.MODE_PRIVATE)
            binding.account.text = sharedPreferences.getString("uUSERNAME",null) + "@" + sharedPreferences.getString("DOMAIN",null)
        }
    }
```

See: https://kotlinlang.org/docs/null-safety.html#safe-calls